resources-manager/include/AdaptivePruningManager.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "IRequestHandler.hpp"
+#include "IResourceManager.hpp"
+#include "InternalBFTClient.hpp"
+
+#include <atomic>
+#include <chrono>
+#include <condition_variable>
+#include <cstdint>
+#include <memory>
+#include <mutex>
+#include <thread>
+
+enum PruningMode { LEGACY, ADAPTIVE };
+
+namespace concord::performance {
+class AdaptivePruningManager {
+ public:
+  AdaptivePruningManager(const std::shared_ptr<concord::performance::IResourceManager> &resourceManager,
```

Comment:
which entity will instantiate the IResourceManager and will handle it to the AdaptivePruningManager? 
Which entity is responsible for the configuration of the feature? 
by the design, I thought that this class, for example, what happens when the operator sends a new configuration for the mapping interval, with this interface, it meant that the AdaptivePruningManager will need to be destroyed and created with a new IResourceManager, instead of managing it..